cmd/system-probe/main_common.go

```diff
@@ -87,20 +92,47 @@ func runAgent() {
 		cleanupAndExit(1)
 	}
 
-	sysprobe, err := CreateSystemProbe(cfg)
-	if err != nil && strings.HasPrefix(err.Error(), ErrSysprobeUnsupported.Error()) {
+	conn, err := net.NewListener(cfg)
+	if err != nil {
+		log.Criticalf("Error creating IPC socket: %s", err)
+		cleanupAndExit(1)
+	}
+
+	loader := NewLoader()
+	httpMux := http.NewServeMux()
+
+	err = loader.Register(cfg, httpMux, factories)
+	if err != nil && strings.HasPrefix(err.Error(), modules.ErrSysprobeUnsupported.Error()) {
 		// If tracer is unsupported by this operating system, then exit gracefully
 		log.Infof("%s, exiting.", err)
 		gracefulExit()
-	} else if err != nil {
+	}
+	if err != nil {
 		log.Criticalf("failed to create system probe: %s", err)
 		cleanupAndExit(1)
 	}
-	defer sysprobe.Close()
+	defer loader.Close()
 
-	go sysprobe.Run()
+	// Register stats endpoint
+	httpMux.HandleFunc("/debug/stats", func(w http.ResponseWriter, req *http.Request) {
+		stats := loader.GetStats()
+		utils.WriteAsJSON(w, stats)
+	})
+
+	http.Serve(conn.GetListener(), httpMux) //nolint:errcheck
```

Comment:
Noticed the `//nolint:errcheck` -  shouldn't we be tracking this error?

Reply:
agreed, I'll add some error handling